shm/src/ship.cpp

```diff
@@ -0,0 +1,115 @@
+#include "ship.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+
+Ship::Ship()
+    : id_(-1) {}
+
+Ship::Ship(int capacity, int maxCrew, int speed, const std::string& name, unsigned int id, Delegate* delegate, Time* time)
+    : capacity_(capacity), maxCrew_(maxCrew), crew_(0), speed_(speed), name_(name), id_(id), payCrew(delegate), time_(time) {
+    time_->addObserver(this);
+}
+
+Ship::Ship(int maxCrew, int speed, unsigned int id, Delegate* delegate, Time* time)
+    : Ship(0, maxCrew, speed, "", id, delegate, time) {}
+
+Ship& Ship::operator-=(size_t num) {
+    if (num > crew_) {
+        crew_ = 0;
+    } else if (crew_ > 0) {
+        crew_ -= num;
+    } else {
+        std::cerr << "There is no crew\n";
+    }
+    return *this;
+}
+
+Ship& Ship::operator+=(size_t num) {
+    if (crew_ < maxCrew_) {
+        crew_ += num;
```

Comment:
I think it should be <= here. If you for example have crew limit 50 and 20 people abroad and you attempt to add 30 more it will fail,because (50 < 50) is false.